src/networking.c

```diff
@@ -405,8 +415,20 @@ void _addReplyToBufferOrList(client *c, const char *s, size_t len) {
      * buffer offset (see function comment) */
     reqresSaveClientReplyOffset(c);
 
+    /* If we're processing a push message into the current client (i.e. executing PUBLISH
+     * to a channel which we are subscribed to, then we wanna postpone that message to be added
+     * after the command's reply (specifically important during multi-exec). the exception is
+     * the SUBSCRIBE command family, which (currently) have a push message instead of a proper reply.
+     * The check for executing_client also avoids affecting push messages that are part of eviction. */
+    if (c == server.current_client && (c->flags & CLIENT_PUSHING) &&
+        server.executing_client && !cmdHasPushAsReply(server.executing_client->cmd))
```

Comment:
Just a little question. If cmd has push as reply doesn't reply push message, i.e. reply with errors, what should we do. For now, I think it is OK as the errors can't happen in multi exec.

Reply:
i'm not sure i understand your question, can you please elaborate?

i will say that these commands are very much broken IMHO, but we can't fix them now.
one example, which might be what you meant, is that we can't say that they don't reply since they can reply with an error.
and in fact their push-type reply does count as a reply, and we do put it as part of the exec array reply.
on the other hand, we can't say that their reply is a push-type, since they can reply with several push messages, not just one (in case they're called with multiple arguments), and that's completely broken (protocol wise, specifically in exec).
anyway, all of that is to be handled in #12207

Reply:
Sorry, I made a mistake.